### ISSUE TITLE
Build failed with type exception

### ISSUE THREAD

**Issue by mirkojahn** [boyney123/eventcatalog#458]
### Have you read the Contributing Guidelines on issues?

- [X] I have read the [Contributing Guidelines on issues](https://github.com/boyney123/eventcatalog/blob/master/CONTRIBUTING.md#reporting-new-issues).

### Description

The issue might be a noob problem, but I couldn't find a way to fix it. Basically the `npm run build --verbose` command fails with the following type exception:
```
> @eventcatalog/core@1.1.0 build
> next build && next export

info  - Linting and checking validity of types ..Failed to compile.

./components/Mdx/SchemaViewer/SchemaViewer.tsx:19:30
Type error: Argument of type '() => any' is not assignable to parameter of type 'never'.

  17 |
  18 | function SchemaViewer({ schema, maxHeight, renderRootTreeLines = false, hideExamples = false, defaultExpandedDepth = 1 }: Props) {
> 19 |   const JsonSchema = useMemo(() => JSON.parse(schema as string), [schema]);
     |                              ^
  20 |
  21 |   return (
  22 |     <JsonSchemaViewer
npm verb exit 1
```

I tried to downgrad till version 1.0.0, but with no success. 
Any help is greatly appreciated.

### Steps to reproduce

I was basically following the Installation guide here: https://www.eventcatalog.dev/docs/installation
1. `npx @eventcatalog/create-eventcatalog@latest my-catalog`
2. `cd ./my-catalog`
3. `npm run build --verbose`

_Side note:_ The command `npm run dev` actually works fine, which let me to believe, that my environment is not totally wrong.

### Expected behavior

successful build

### Actual behavior

```
> @eventcatalog/core@1.1.0 build
> next build && next export

info  - Linting and checking validity of types ..Failed to compile.

./components/Mdx/SchemaViewer/SchemaViewer.tsx:19:30
Type error: Argument of type '() => any' is not assignable to parameter of type 'never'.

  17 |
  18 | function SchemaViewer({ schema, maxHeight, renderRootTreeLines = false, hideExamples = false, defaultExpandedDepth = 1 }: Props) {
> 19 |   const JsonSchema = useMemo(() => JSON.parse(schema as string), [schema]);
     |                              ^
  20 |
  21 |   return (
  22 |     <JsonSchemaViewer
npm verb exit 1
```

### Your environment

- EventCatalog version used: 
  - 1.10.0
- Environment name and version (e.g. Chrome 89, Node.js 16.4):
  - Node: v20.9.0
  - npm/ npx: 10.1.0
- Operating system and version (e.g. Ubuntu 20.04.2 LTS):
  - 22.04.3 LTS (Jammy Jellyfish) - in a WSL
  - 5.15.90.1-microsoft-standard-WSL2

My package json looks like this:
```
{
  "name": "my-catalog",
  "version": "0.0.1",
  "private": true,
  "scripts": {
    "start": "eventcatalog start",
    "dev": "eventcatalog dev",
    "build": "eventcatalog build",
    "generate": "eventcatalog generate",
    "test": "echo \"Error: no test specified\" && exit 1"
  },
  "devDependencies": {
    "@eventcatalog/types": "0.4.2",
    "@types/node": "20.8.8",
    "@types/react": "18.2.32",
    "autoprefixer": "10.4.5",
    "cross-env": "^7.0.3",
    "postcss": "^8.3.11",
    "tailwindcss": "^3.3.3",
    "typescript": "^5.1.6"
  },
  "dependencies": {
    "@eventcatalog/core": "1.10.0"
  }
}
```

**Comment by mirkojahn** [comment]
I don't know why, bot doing everything over again solved the issue. Not sure what was the cause :-/

**Comment by 1bberto** [comment]
that is really weird @mirkojahn, cuz I'm currently facing this issue on my pipeline

**Comment by mirkojahn** [comment]
I think it was related to the users in the `eventcatalog.config.js` file. I played with that and replaced the provided users. I really don't know, if that was related, but that would be a place I would start looking into. I reverted every change in my project (except that file) and still got the error. After recreating the project and having just that one change... it worked. There might have been other stuff involved, so take that with a grain of salt, but that is all I have. Hope that helps.
